boards: phytec: add support for phyBOARD-Nash (imx93)

jonas-rem opened the pull request:
~Depends on #76829~

Add basic support for the phyBOARD-Nash, an SBC based on the phyCORE-i.MX93. Both CPU types (Cortex-A55, Cortex-M33) are supported.

nordicjm commented on boards/phytec/mimx93_phyboard_nash/CMakeLists.txt:
```diff
@@ -0,0 +1 @@
+# SPDX-License-Identifier: Apache-2.0
```
no empty files

jonas-rem replied:
fixed

nordicjm commented on boards/phytec/mimx93_phyboard_nash/doc/img/phyboard_nash.webp:
reduce this file size by a lot on https://tinypng.com/

nordicjm replied:
@kartben can we remove allowing jpegs and enforce webps and also reduce the max allowed file size?

jonas-rem replied:
Tinypng produced an image that was still over 500 kB. But converting via the command line tool `cwebp` (libwebp) works well (150 kB now).

`cwebp -q 40 in.png -o out.webp`

kartben replied:
> Tinypng produced an image that was still over 500 kB. But converting via the command line tool `cwebp` (libwebp) works well (150 kB now).
> 
> `cwebp -q 40 in.png -o out.webp`

you do not need a 1400px wide image for basic illustrative purposes :) You may do the following on your original hi-res "raw" picture:

```
cwebp -resize 650 0 ~/Downloads/phyboard_nash.webp -o out.webp
```

jonas-rem replied:
650 px would not be enough to be able to read the silkscreen when zooming in. But 1000 px at a quality of 40 is sufficient. This reduces the size to 84 kByte. Deal? ;)

nordicjm commented on boards/phytec/mimx93_phyboard_nash/mimx93_phyboard_nash_a55.dts:
```diff
@@ -0,0 +1,40 @@
+
+	cpus {
+		cpu@0 {
+			status = "disabled";
```
why is cpu0 disabled?

jonas-rem replied:
The imx93 variant on the phyBOARD-Nash has 2 A55 CPUs. I guess the default use-case would be to run Zephyr on one core for specific real-time applications and not to use SMP. NXP usually disabled all but one core in their Cortex-A EVK boards.

jonas-rem commented:
Rebased on current main after #76829 is merged.

nordicjm commented on boards/phytec/mimx93_phyboard_nash/doc/index.rst:
```diff
@@ -0,0 +1,182 @@
+Supported Features
+==================
+
+The phyBOARD Nash Cortex-A Core configuration supports the following hardware
```
should have the exact name used to build using west/cmake and label as board target not board configuration

jonas-rem replied:
Right, that gives a better overview, I changed that.

nordicjm commented on boards/phytec/mimx93_phyboard_nash/doc/index.rst:
```diff
@@ -0,0 +1,182 @@
+| TPM       | on-chip    | TPM Counter                         |
++-----------+------------+-------------------------------------+
+
+The phyBOARD Nash Cortex-M33 configuration supports the following hardware
```
as above

nordicjm commented on boards/phytec/mimx93_phyboard_nash/mimx93_phyboard_nash_a55_defconfig:
```diff
@@ -0,0 +1,31 @@
+CONFIG_UART_CONSOLE=y
+
+CONFIG_CLOCK_CONTROL=y
+CONFIG_PINCTRL=y
```
pinctrl goes in drivers not board files

jonas-rem replied:
fixed

nordicjm commented on boards/phytec/mimx93_phyboard_nash/mimx93_phyboard_nash_m33_defconfig:
```diff
@@ -0,0 +1,10 @@
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_CLOCK_CONTROL=y
+CONFIG_PINCTRL=y
```
as above

jonas-rem replied:
fixed

jonas-rem commented:
Thanks for the review @nordicjm, I applied all your hints.

nordicjm commented:
Ping @kartben

dnltz commented on boards/phytec/mimx93_phyboard_nash/mimx93_phyboard_nash_a55.dts:
```diff
@@ -0,0 +1,40 @@
+	/* clocks = <&ccm IMX_CCM_UART4_CLK 0x6c 24>; */
+	pinctrl-0 = <&uart2_default>;
+	pinctrl-names = "default";
+};
```
GPIOs are not enabled but listed as supported feature. Either remove from docs or enable them here.

jonas-rem replied:
Removed GPIOs from doc and the *.yaml files. If there is more time for testing, I will add those features step-by-step in follow-up PRs.

dnltz commented on boards/phytec/mimx93_phyboard_nash/mimx93_phyboard_nash_a55.yaml:
```diff
@@ -0,0 +1,21 @@
+supported:
+  - gpio
+  - uart
+  - i2c
+  - spi
```
They're not supported right now.

jonas-rem replied:
removed

dnltz commented on boards/phytec/mimx93_phyboard_nash/mimx93_phyboard_nash_a55_defconfig:
```diff
@@ -0,0 +1,30 @@
+#
+# Copyright 2022 NXP
+#
```
Add PHYTEC here.

jonas-rem replied:
I changed the copyright notice to PHYTEC, as those are just some board related configuration flags.

dnltz commented on boards/phytec/mimx93_phyboard_nash/mimx93_phyboard_nash_m33.dts:
```diff
@@ -0,0 +1,31 @@
+	current-speed = <115200>;
+	pinctrl-0 = <&uart2_default>;
+	pinctrl-names = "default";
+};
```
Same story with GPIOs here.

dnltz commented on boards/phytec/mimx93_phyboard_nash/board.yml:
```diff
@@ -0,0 +1,5 @@
+board:
+  name: mimx93_phyboard_nash
```
I would suggest to remove `mimx93` from the board name. SoC vendors normally call their EVM, EVK, ... after the silicon itself to better promote it. We have unique names for our boards (`phyBOARD-Nash`).

I also removed them from our TI boards since hardware model v2 allows multiple SoC in one board directory. It just looks cleaner IMO:

```
phyboard_nash/mimx9352/a55
phyboard_nash/mimx9352/m33
```


```suggestion
  name: phyboard_nash
```

jonas-rem replied:
I followed the pattern that we used for the imx8 SoCs but I agree that it looks cleaner to omit the prefix, now that there is HWM2.

Will can a PR in the next days that aligns the naming for the PHYTEC imx8 platforms.

dnltz replied:
Thanks!

jonas-rem replied:
As reference I'll mention the PR here: https://github.com/zephyrproject-rtos/zephyr/pull/79575

jonas-rem commented on boards/phytec/mimx93_phyboard_nash/Kconfig.defconfig:
```diff
@@ -0,0 +1,45 @@
+
+if BOARD_MIMX93_PHYBOARD_NASH_MIMX9352_A55
+
+if NETWORKING
```
Right, this is not yet supported. As there are no board specific components, I am removing `Kconfig.defconfig` completely.

Commit:
boards: phytec: add support for phyBOARD-Nash (imx93)

Add basic support for the phyBOARD-Nash, a SBC based on the phyCORE-i.MX93.
Both CPU types (Cortex-A55, Cortex-M33) are supported.

Signed-off-by: Jonas Remmert <j.remmert@phytec.de>

dnltz commented on boards/phytec/phyboard_nash/doc/index.rst:
```diff
@@ -0,0 +1,178 @@
+.. _phyboard_nash:
+
+PhyBOARD Nash (NXP i.MX93)
```
```suggestion
phyBOARD-Nash (NXP i.MX93)
```

dnltz replied:
As you can see looks a little bit odd here: https://docs.zephyrproject.org/latest/boards/phytec/index.html

You could also just called it `phyBOARD-Nash i.MX93`

dnltz commented on boards/phytec/phyboard_nash/doc/index.rst:
```diff
@@ -0,0 +1,178 @@
+Overview
+********
+
+The phyBOARD Nash is based on the phyCORE-i.MX93 SoM is based on the NXP i.MX93
```
```suggestion
The phyBOARD-Nash is based on the phyCORE-i.MX93 SoM is based on the NXP i.MX93
```

dnltz commented on boards/phytec/phyboard_nash/phyboard_nash_m33.yaml:
```diff
@@ -0,0 +1,15 @@
+# SPDX-License-Identifier: Apache-2.0
+
+identifier: phyboard_nash/mimx9352/m33
+name: Phytec phyBOARD Nash (i.MX93) M33
```
```suggestion
name: PHYTEC phyBOARD-Nash (i.MX93) M33
```

dnltz commented on boards/phytec/phyboard_nash/phyboard_nash_a55.yaml:
```diff
@@ -0,0 +1,18 @@
+# SPDX-License-Identifier: Apache-2.0
+
+identifier: phyboard_nash/mimx9352/a55
+name: Phytec phyBOARD Nash (i.MX93) A55
```
```suggestion
name: PHYTEC phyBOARD-Nash (i.MX93) A55
```

jonas-rem replied:
fixed

dnltz commented on boards/phytec/phyboard_nash/doc/index.rst:
```diff
@@ -0,0 +1,178 @@
+amount of resources in more specific designs.
+
+i.MX93 MPU is composed of one cluster of 2x Cortex-A55 cores and a single
+Cortex®-M33 core. Zephyr OS is ported to run on one of the Cortex®-A55 core as
```
I would remove these trademark symbols here. You're not consistent with them and I don't see any requirement here.

jonas-rem replied:
Yes, removed it.

dnltz reviewed (changes requested):
There are some minor issues I found. Everything else looks good!

dnltz commented:
@jonas-rem did you see those errors?

jonas-rem commented:
> @jonas-rem did you see those errors?

@dnltz I saw but could not reproduce them locally. Rebased on current main to restart the CI pipeline, now it doesn't show any error.

jonas-rem commented on boards/phytec/phyboard_nash/board.yml:
```diff
@@ -0,0 +1,6 @@
+board:
+  name: phyboard_nash
+  full_name: phyBOARD-Nash i.MX93
```
Just added the full_name property to the board as #79571 has been merged now.

dnltz reviewed (approved):
Thanks :)

Please update the release notes after #79575 got merged. I'm fine with another pull-request to not block this issue, because you might run into a merge conflict.
